code/datums/repositories/crew/crew.dm

```diff
@@ -49,12 +49,12 @@ var/global/datum/repository/crew/crew_repository = new()
 
 	cache_data_alert[num2text(z_level)] = FALSE
 	var/tracked = scan()
-	for(var/obj/item/weapon/rig/R in tracked)
-		var/turf/pos = get_turf(R)
-		if(R.tracking_level != RIG_SENSOR_OFF && pos?.z == z_level)
+	for(var/obj/item/rig_module/healthbar/HB in tracked)
+		var/turf/pos = get_turf(HB)
+		if(HB.tracking_level != RIG_SENSOR_OFF && pos?.z == z_level && HB.holder.active)
 
-			var/list/crewmemberData = list("sensor_type"=R.tracking_level, "stat"=R.wearer.stat, "area"="", "x"=-1, "y"=-1, "z"=-1, "ref"="\ref[R.wearer]")
-			if(!(run_queues(R.wearer, R, pos, crewmemberData) & MOD_SUIT_SENSORS_REJECTED))
+			var/list/crewmemberData = list("sensor_type"=HB.tracking_level, "stat"=HB.holder.wearer.stat, "area"="", "x"=-1, "y"=-1, "z"=-1, "ref"="\ref[HB.holder.wearer]")
+			if(!(run_queues(HB.holder.wearer, HB, pos, crewmemberData) & MOD_SUIT_SENSORS_REJECTED))
```

Comment:
need to account for the possibility that the rig isn't being worn, or the much rarer possibility that the module isnt installed in a rig, make sure we dont get runtime errors here

Probably the tracker should register when someone puts on the rig for the first time

Reply:
it's already handled in scan()